deepwell/src/endpoints/parent.rs

```diff
@@ -85,3 +88,90 @@ pub async fn parent_remove(
 
     ParentService::remove(ctx, input).await
 }
+
+pub async fn parent_get_all(
+    ctx: &ServiceContext<'_>,
+    params: Params<'static>,
+) -> Result<Vec<String>> {
+    let GetPageReference { site_id, page } = params.parse()?;
+
+    info!(
+        "Getting parents for child {:?} in site ID {}",
+        page, site_id,
+    );
+
+    let parents: Vec<Reference<'_>> = ParentService::get_parents(ctx, site_id, page)
+        .await?
+        .iter()
+        .map(|p| Reference::from(p.parent_page_id))
+        .collect();
+
+    let pages: Vec<String> = PageService::get_pages(ctx, site_id, parents.as_slice())
+        .await?
+        .iter()
+        .map(|p| p.slug.clone())
+        .collect();
+
+    Ok(pages)
+}
+
+pub async fn parent_modify(
+    ctx: &ServiceContext<'_>,
+    params: Params<'static>,
+) -> Result<ModifyParentOutput> {
+    let input: ParentModifyDescription = params.parse()?;
+
+    info!(
+        "Modifying multiple parental relationship for child {:?} in site ID {}",
+        input.child, input.site_id,
+    );
+
+    let creation = match input.add {
+        Some(parents) => {
+            let mut creation = Vec::new();
+            for parent in parents {
+                if let Ok(Some(model)) = ParentService::create(
+                    ctx,
+                    ParentDescription {
+                        site_id: input.site_id,
+                        parent: parent.clone(),
+                        child: input.child.clone(),
+                    },
+                )
+                .await
```

Comment:
This ignores any failures. Shouldn't the whole call fail if one of the constituent parts fails?

Reply:
Would `try_join_all` revert all the parental relation creation or deletion if one of them failed? Otherwise the update would be half done, and there are no guarantees which ones would be added or deleted even if they could succeed.

Reply:
In deepwell, all methods are wrapped in a transaction (it's what's inside the `ctx`). So if something fails then all DB changes will be reverted. Whether you use `try_join!()` (though since this is a dynamic list of futures it would need to be a [`JoinSet`](https://docs.rs/tokio/latest/tokio/task/join_set/struct.JoinSet.html) I guess) or run them one after another is not relevant.

The issue I was bringing attention to here is, because no error is raised (it is silenced), the transaction will go through and commit. If it gets passed up the chain then the transaction will get rolled back.